vpr/src/device/rr_graph_view.h

```diff
@@ -62,6 +62,21 @@ class RRGraphView {
         return node_storage_.node_capacity(node);
     }
 
+    /* Get the direction of a routing resource node. This function is inlined for runtime optimization.
+     * INC_DIRECTION: wire driver is positioned at the low-coordinate end of the wire.
+     * DEC_DIRECTION: wire_driver is positioned at the high-coordinate end of the wire.
+     * BI_DIRECTION: wire has multiple drivers, so signals can travel either way along the wire
+     * INVALID_DIRECTION: wire does not have a valid direction set
+     */
+    inline e_direction node_direction(RRNodeId node) const {
```

Comment:
I looked at the needs of the node_direction. 
Most of the time, client functions want to know if this is a bi-directional routing track or not. In some cases, it wants to know the directionality for uni-directional tracks. 
As @vaughnbetz suggested, we should avoid exposing the raw data to developers.
Should we think about creating a query method ``node_is_bidirectional()`` and ``node_is_unidirectional()``? For unidirectional wires, we can return the ``INC_DIR`` and ``DEC_DIR`` data to developers.

I am also happy with the current APIs. These are just suggestions.

Reply:
Making a node_is_bidirectional() call makes sense to me, as the number one use of these calls seems to be to check if the node is bidirectional.  I don't have a really strong opinion though; if you want to land it as is that's OK with me, but adding a node_is_bidirectional method and using it instead of == BI_DIRECTION may add a small amount of abstraction so I'm also fine with changing to that.